src/inmanta/server/bootloader.py

```diff
@@ -82,7 +83,16 @@ async def start(self) -> None:
         await self.restserver.start()
         self.started = True
 
-    async def stop(self) -> None:
+    async def stop(self, timeout: Optional[int] = None) -> None:
+        """
+        :param timeout: Raises TimeoutError when the server hasn't finished stopping after
+                        this amount of seconds.
+        """
+        if not timeout:
+            await self._stop()
+        await asyncio.wait_for(self._stop(), timeout=timeout)
```

Comment:
this line will be always executed  (i.e. there is no else)

Reply:
Wow. Indeed. How did I miss that?